src/Controllers/AuthController.cs

```diff
@@ -0,0 +1,134 @@
+using System.Text.Encodings.Web;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using server.Context;
+using server.Dtos;
+using server.Dtos.User;
+using server.Models;
+using server.Services;
+using server.Validators;
+
+namespace Server.Controllers;
+
+public class AuthController : ApiControllerBase
+{
+    private readonly ApplicationDbContext _dbContext;
+    private readonly IEmailService _emailService;
+
+    public AuthController(ApplicationDbContext dbContext, IEmailService emailService)
+    {
+        _dbContext = dbContext;
+        _emailService = emailService;
+    }
+
+    [HttpPost("register")]
+    public async Task<ActionResult<User>> Register(CreateUserDto createUserDto)
+    {
+        User user = new User
+        {
+            Name = createUserDto.Name,
+            Email = createUserDto.Email,
+            PasswordHash = BCrypt.Net.BCrypt.HashPassword(createUserDto.Password),
+            EmailConfirmationToken = "Token" // todo 
+        };
+
+        UserValidator validator = new UserValidator();
```

Comment:
Why are you validating the dto by hand? Create a middleware. I think fluent validation has a package for this: `FluentValidation.AspNetCore` something like that